docs/agent-development.md

```diff
@@ -334,7 +336,64 @@ In addition to each agent having a "native" API for instrumentation, they also i
 
 ### Distributed Tracing
 
-We implement the [W3C Trace Context](https://w3c.github.io/trace-context/) standard, with the exceptions that we currently use the header name "elastic-apm-traceparent" and we do not currently support "tracestate". Both of these exceptions will be removed as Trace Context nears finalisation; we are tracking this in https://github.com/elastic/apm/issues/71.
+We implement the W3C standards, both for HTTP headers and binary fields.
+
+#### Http Headers
+
+Our implementation relies on the [W3C Trace Context](https://www.w3.org/TR/trace-context-1/) standard. Until this standard became final
+, we used the header name `elastic-apm-traceparent` and we did not support `tracestate`. Soon after the standard became official, we
+started to fully align with it. For backward compatibility reasons, this was done in phases, so that the first step was to look for both
+`traceparent` headers in incoming requests and sending both `traceparent` headers in outgoing requests (with the exception of the RUM agent
+, due to CORS). [Issue #71](https://github.com/elastic/apm/issues/71) describes this in more detail, as well as tracking implementation
+in the different agents.
+New agents may decide whether to support both `traceparent` headers (so to be compatible with older agent versions) or only the formal W3C
+header.
+
+#### Binary Fields
+
+Our implementation relies on the [W3C Binary Trace Context](https://w3c.github.io/trace-context-binary/) standard. Since we started
+implementing it when this was still a draft, we named the field `elasticapmtraceparent` instead of `traceparent`, and we decided to
+wait with the implementation of the `tracestate` field. In order to make sure we are fully aligned, we took the following "snapshot" of the
+ draft which is followed by all agents implementing it:
+
+**_elasticapmtraceparent binary format_**
```

Comment:
Maybe just add a permalink to the w3c repo rather than copying over?: https://github.com/w3c/trace-context-binary/blob/571cafae56360d99c1f233e7df7d0009b44201fe/spec/20-binary-format.md

Reply:
Good point, as long as you point to a SHA (as Felix has done) it won't change.

Reply:
Good idea, thanks!